deepchecks/vision/base.py

```diff
@@ -147,90 +130,35 @@ def task_type(self) -> TaskType:
             self._task_type = task_type_check(self.model, self.train)
         return self._task_type
 
+    @property
+    def train_sample_predictions(self):
+        """Return the predictions on the train samples."""
+        if self._train_sample_predictions is None:
+            self._train_sample_predictions = []
+            for tensor, _ in self.train.sample_data_loader:
+                self._train_sample_predictions.append(self.model(tensor))
+        return self._train_sample_predictions
+
+    @property
+    def test_sample_predictions(self):
+        """Return the predictions on the test samples."""
+        if self._test_sample_predictions is None:
+            self._test_sample_predictions = []
+            for tensor, _ in self.test.sample_data_loader:
+                self._test_sample_predictions.append(self.model(tensor))
+        return self._test_sample_predictions
+
     def have_test(self):
         """Return whether there is test dataset defined."""
         return self._test is not None
 
     def assert_task_type(self, *expected_types: TaskType):
-        """Assert task_type matching given types.
-
-        If task_type is defined, validate it and raise error if needed, else returns True.
-        If task_type is not defined, return False.
-        """
-        # To calculate task type we need model and train. if not exists return False, means we did not validate
-        if self._model is None or self._train is None:
-            return False
```

Comment:
Why did you remove this?

Reply:
a. it goes nowhere the return, we use this method as "raise error else it's okay"
b. it is already encapsulated when calling `task_type`. I'm not sure about the method there, if we can have task_type without model than the validation there also need to change